packages/agent/app/create-harness.ts

```diff
@@ -47,14 +49,31 @@ function *runTest(parentFrame: ParentFrame, test: TestImplementation, context: T
     try {
       console.debug('[harness] running step', step);
       parentFrame.send({ type: 'step:running', path: stepPath });
-      let result: TestContext | void = yield step.action(context);
+
+      let result: TestContext | void = yield timebox(step.action(context), 2000)
+
       if (result != null) {
         context = {...context, ...result};
       }
       parentFrame.send({ type: 'step:result', status: 'ok', path: stepPath });
     } catch(error) {
       console.error('[harness] step failed', step, error);
-      parentFrame.send({ type: 'step:result', status: 'failed', error: serializeError(error), path: stepPath });
+      if (error.name === 'TimeoutError') {
+        parentFrame.send({
+          type: 'step:result',
+          status: 'failed',
+          timeout: true,
+          path: stepPath
+        })
+      } else {
+        parentFrame.send({
+          type: 'step:result',
+          status: 'failed',
+          timeout: false,
+          error: serializeError(error),
+          path: stepPath
+        });
+      }
```

Comment:
Didn't realize this was originally introduced here, I have the same comment as on #293: can we do this by racing the timeout against the success case instead? The try/catch seems a bit awkward to me.